ray-operator/apis/ray/v1alpha1/rayservice_types.go

```diff
@@ -1,22 +1,85 @@
 package v1alpha1
 
 import (
+	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
+type ServiceStatus string
+
+const (
+	FAIL_TO_GET_RAYSERVICE           ServiceStatus = "fail_to_get_rayservice"
+	FAIL_TO_GET_OR_CREATE_RAYCLUSTER ServiceStatus = "fail_to_get_or_create_RAYCLUSTER"
+	WAIT_FOR_DASHBOARD               ServiceStatus = "wait_for_dashboard"
+	FAIL_SERVE_DEPLOY                ServiceStatus = "fail_serve_deploy"
+	FAIL_GET_SERVE_DEPLOYMENT_STATUS ServiceStatus = "fail_get_serve_deployment_status"
+	RUNNING                          ServiceStatus = "running"
+	RESTARTING                       ServiceStatus = "restarting"
+	FAIL_DELETE_RAYCLUSTER           ServiceStatus = "fail_delete_raycluster"
+)
+
+// ServeConfigSpec defines the desired state of RayService
+type ServeConfigSpec struct {
+	Name                      string             `json:"name"`
+	ImportPath                string             `json:"import_path"`
+	InitArgs                  []string           `json:"init_args,omitempty"`
+	InitKwargs                map[string]string  `json:"init_kwargs,omitempty"`
+	NumReplicas               *int32             `json:"num_replicas,omitempty"`
```

Comment:
Replicas is slightly overloaded -- would be redundant to call it `NumServeReplicas`?

Reply:
These names are copied from [the Serve deployment options](https://docs.ray.io/en/latest/serve/package-ref.html#ray.serve.api.Deployment.options), and this spec is applied to the Ray cluster via the Serve REST API ([Pydantic schema](https://github.com/ray-project/ray/blob/16bdfe6a3981a08a02a66984dadeac5b73c7cb1f/python/ray/serve/schema.py#L81-L201)), which uses these names. I think it's probably okay to keep the names as they are since they match the deployment options that Serve users are familiar with. What do you think?

Reply:
Ok -- leaving the name as is sounds good.

I do think we should stick to K8s-style camelCase for the json names of the fields.

Reply:
Sounds good. @brucez-anyscale Would camel-casing the JSON mean that the REST API needs to accept camel-cased fields? I can update the REST API in that case.

Reply:
I think it would be better to just do the conversion in the operator code.
It's silly that K8s conventions vs. Python conventions create extra work, but that's life.

Reply:
oh, so we want to do a conversion before we send it to REST API.
We could do it though.